src/components/course/EnrollModal.jsx

```diff
@@ -1,55 +1,206 @@
+  enterpriseOfferUsageWithPrice: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.text.enterprise-offer-usage.with-price',
+    defaultMessage: 'You are about to redeem {courseRunPrice} from your learner credit. This action cannot be reversed.',
+    description: 'Text for the confirmation modal when using an enterprise offer with a price',
+  },
+  enterpriseOfferUsageWithoutPrice: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.text.enterprise-offer-usage.without-price',
+    defaultMessage: 'You are about to redeem your learner credit for this course. This action cannot be reversed.',
+    description: 'Text for the confirmation modal when using an enterprise offer with a limit',
+  },
+  upgradeCoveredByOrg: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.text.upgrade-covered-by-org',
+    defaultMessage: 'This course is covered by your organization, which allows you to upgrade for free.',
+    description: 'Text for the confirmation modal when upgrading is covered by the organization',
+  },
+  upgradeBenefitsPrefix: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.list-items.prefix',
+    defaultMessage: 'By upgrading, you will get:',
+    description: 'Prefix for the list of benefits of upgrading',
+  },
+  upgradeBenefitsUnlimitedAccess: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.list-items.unlimited-access',
+    defaultMessage: 'Unlimited access to course materials',
+    description: 'List item for the benefits of upgrading, including unlimited access.',
+  },
+  upgradeBenefitsFeedbackAndGradedAssignments: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.list-items.feedback-graded-assignments',
+    defaultMessage: 'Feedback and graded assignments',
+    description: 'List item for the benefits of upgrading, including feedback.',
+  },
+  upgradeBenefitsShareableCertificate: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.list-items.shareable-certificate',
+    defaultMessage: 'Shareable certificate upon completion',
+    description: 'List item for the benefits of upgrading, including a shareable certificate.',
+  },
+  confirmationCtaLoading: {
+    id: 'enterprise.learner_portal.enroll-upgrade-modal.buttons.loading',
+    defaultMessage: 'Loading...',
+    description: 'Text for the confirmation button in the modal when loading',
+  },
+});
 
-export const createUseEnterpriseOfferText = (offer, courseRunPrice) => {
-  if (offer.offerType === ENTERPRISE_OFFER_TYPE.ENROLLMENTS_LIMIT) {
-    return 'You are about to redeem 1 learner credit. This action cannot be reversed.';
+export const createUseEnterpriseOfferText = (offerType, courseRunPrice) => {
+  if (offerType !== ENTERPRISE_OFFER_TYPE.ENROLLMENTS_LIMIT && courseRunPrice) {
+    return messages.enterpriseOfferUsageWithPrice;
   }
-  return `You are about to redeem $${courseRunPrice} from your learner credit. This action cannot be reversed.`;
+  return messages.enterpriseOfferUsageWithoutPrice;
+};
```

Comment:
[nit/curious] Just wondering if this will take into account the enterprise customer flag for disabling price visibility? Should we take it into account here?

Reply:
@brobro10000 Great question/catch. I agree it likely _should_ take into that customer flag for disabling price visibility. That said, I will defer on this issue for a standalone, follow-up PR specific to it :)

Reply:
@brobro10000 Follow-up PR for the above feedback: https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/1117